SciGraph-core/src/main/java/io/scigraph/internal/TinkerGraphUtil.java

```diff
@@ -157,32 +189,32 @@ static Edge addEdge(Graph graph, Edge edge) {
   }
 
 
-  public static Element addElement(Graph graph, Element element) {
+  public Element addElement(Element element) {
     if (element instanceof Vertex) {
-      return addNode(graph, (Vertex) element);
+      return addNode((Vertex) element);
     } else {
-      return addEdge(graph, (Edge) element);
+      return addEdge((Edge) element);
     }
   }
 
-  public static void addGraph(Graph graph, Graph addition) {
+  public void addGraph(Graph addition) {
     for (Vertex vertex : addition.getVertices()) {
-      addElement(graph, vertex);
+      addElement(vertex);
     }
     for (Edge edge : addition.getEdges()) {
-      addElement(graph, edge);
+      addElement(edge);
     }
   }
 
-  public static Graph combineGraphs(Graph graph1, Graph graph2) {
-    Graph graph = new TinkerGraph();
-    addGraph(graph, graph1);
-    addGraph(graph, graph2);
-    return graph;
+  public Graph combineGraphs(Graph graph2) {
+    TinkerGraphUtil tgu = new TinkerGraphUtil(curieUtil);
+    tgu.addGraph(graph);
+    tgu.addGraph(graph2);
+    return tgu.getGraph();
   }
 
-  public static TinkerGraph resultToGraph(Result result) {
-    TinkerGraph graph = new TinkerGraph();
+  public Graph resultToGraph(Result result) {
```

Comment:
I think that it makes sense to have it as static, this is what I had in mind, let me know if it makes sense:

```
  public Graph resultToGraph(Result result) {
    TinkerGraph resultGraph = new TinkerGraph();
    while (result.hasNext()) {
      Map<String, Object> map = result.next();
      for (Map.Entry<String, Object> entry : map.entrySet()) {
        Object value = entry.getValue();
        String key = entry.getKey();
        if (null == value) {
          continue;
        } else if (value instanceof PropertyContainer) {
          resultGraph.addElement((PropertyContainer) value);
        } else if (value instanceof Path) {
          for (PropertyContainer container : (Path) value) {
            resultGraph.addElement(container);
          }
        } else if (value instanceof SeqWrapper) {
          for (Object thing : (SeqWrapper<?>) value) {
            if (thing instanceof PropertyContainer) {
              resultGraph.addElement((PropertyContainer) thing);
            }
          }
        } else if (value instanceof Boolean) {
          // generates a lonely node which contains the result
          Vertex vertex = resultGraph.addVertex(key);
          vertex.setProperty(key, value);
          vertex.setProperty(NodeProperties.LABEL, "Boolean result");
          vertex.setProperty(CommonProperties.IRI, key);
        } else {
          logger.warning("Not converting " + value.getClass() + " to tinker graph");
        }
      }
    }
    return resultGraph;
  }
```